examples/search_bar/main.rs

```diff
@@ -0,0 +1,33 @@
+use gtk::prelude::*;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.search_bar"),
+        Default::default(),
+    );
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+    window.set_title(Some("Search Bar"));
+
+    let search_bar = gtk::SearchBar::new();
+    search_bar.set_valign(gtk::Align::Start);
+    window.set_child(Some(&search_bar));
+    search_bar.set_key_capture_widget(Some(&window));
+
+    let search_box = gtk::Box::new(gtk::Orientation::Horizontal, 6);
+    search_bar.set_child(Some(&search_box));
+
+    let entry = gtk::SearchEntry::new();
+    entry.set_hexpand(true);
+    search_box.append(&entry);
+    search_bar.connect_entry(&entry);
+
+    let menu_button = gtk::MenuButton::new();
+    search_box.append(&menu_button);
```

Comment:
Is this button really useful for this demo?

Reply:
No, it just demonstrates how one might be added to a `SearchBar` I assume.

Reply:
What would be interesting to do here is have a header bar with a `gtk::ToggleButton` that would update SearchEntry's `is_search_mode` with the ToggleButton's state

Reply:
Okay I'll do that

Reply:
Finished!